### PR TITLE
Initial porting

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@gsvr review please. This is mostly a simplification of the code from mama-ng-control that you already extensively reviewed

File: message_sender/tasks.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    kwargs = dict(target=target, payload=payload,
+                  instance_id=instance_id, hook_id=hook.id)
+    DeliverHook.apply_async(kwargs=kwargs)
+
+
+class Send_Metric(Task):
+
+    """
+    Task to fire metrics
+    TODO: Replace fire with metrics client creation when ready.
```
Comment (gsvr) [inline message_sender/tasks.py:51]: Do you want to make open an issue and link it here?

File: message_sender/tasks.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    def run(self, metric, value, agg, **kwargs):
+        """
+        Returns count from api
+        """
+        l = self.get_logger(**kwargs)
+
+        l.info("Firing metric: %r [%s] -> %g" % (metric, agg, float(value)))
+        try:
+            # TODO: Real metric firing
```
Comment (gsvr) [inline message_sender/tasks.py:77]: Same - issue number?

File: message_sender/tests.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    def check_logs(self, msg):
+        if self.handler.logs is None:  # nothing to check
+            return False
+        if type(self.handler.logs) != list:
+            [logs] = self.handler.logs
+        else:
+            logs = self.handler.logs
+        for log in logs:
+            logline = log.msg.replace("u'", "'")
+            print(logline)
```
Comment (gsvr) [inline message_sender/tests.py]: stray `print`

File: message_sender/tests.py [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+
+        c = Outbound.objects.get(pk=existing)
+        self.assertEqual(c.delivered, False)
+        self.assertEqual(c.attempts, 2)
+        self.assertEqual(c.metadata["nack_reason"],
+                         "no answer")
+        self.assertEquals(True, self.check_logs(
+            "Message: 'Simple outbound message' sent to '+27123' "
+            "[session_event: new]"))
+        # TODO: Bring metrics back
```
Comment (gsvr) [inline message_sender/tests.py:324]: issue number?

File: message_sender/tests.py [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+
+        d = Outbound.objects.get(pk=failed.id)
+        self.assertEqual(d.delivered, False)
+        self.assertEqual(d.attempts, 3)  # not moved on as last attempt passed
+        self.assertEqual(d.metadata["nack_reason"],
+                         "no answer")
+        self.assertEquals(False, self.check_logs(
+            "Message: 'Simple outbound message' sent to '+27123'"
+            "[session_event: new]"))
+        # TODO: Bring metrics back
```
Comment (gsvr) [inline message_sender/tests.py:364]: issue number

File: message_sender/views.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                status = 400
+                accepted = {"accepted": False,
+                            "reason": "Missing expected body keys"}
+        except ObjectDoesNotExist:
+            status = 400
+            accepted = {"accepted": False,
+                        "reason": "Missing message in control"}
+        return Response(accepted, status=status)
+
+    # TODO make this work in test harness, works in production
```
Comment (gsvr) [inline message_sender/views.py:106]: maybe an issue number here, but I guess this is a general problem that we won't forget about.

**Comment by gsvr** [comment]
Minor comments, :+1:

**Pull request merged**